bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/LeapBridgeHandler.java

```diff
@@ -0,0 +1,763 @@
+            return;
+        } catch (IllegalArgumentException e) {
+            // port out of valid range
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Invalid port number");
+            return;
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error opening SSL connection. Check log.");
+            logger.info("Error opening SSL connection: {}", e.getMessage());
+            disconnect();
+            scheduleConnectRetry(reconnectInterval); // Possibly a temporary problem. Try again later.
+            return;
+        }
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, STATUS_INITIALIZING);
+
+        Thread readerThread = new Thread(this::readerThreadJob, "Lutron reader");
+        readerThread.setDaemon(true);
+        readerThread.start();
+        this.readerThread = readerThread;
+
+        Thread senderThread = new Thread(this::senderThreadJob, "Lutron sender");
+        senderThread.setDaemon(true);
+        senderThread.start();
+        this.senderThread = senderThread;
+
+        sendCommand(new LeapCommand(Request.getButtonGroups()));
+        queryDiscoveryData();
+        sendCommand(new LeapCommand(Request.subscribeOccupancyGroupStatus()));
+
+        logger.debug("Starting keepAlive job with interval {}", heartbeatInterval);
+        keepAlive = scheduler.scheduleWithFixedDelay(this::sendKeepAlive, heartbeatInterval, heartbeatInterval,
+                TimeUnit.MINUTES);
+    }
+
+    /**
+     * Called by connect() and discovery service to request fresh discovery data
+     */
+    public void queryDiscoveryData() {
+        sendCommand(new LeapCommand(Request.getDevices()));
+        sendCommand(new LeapCommand(Request.getAreas()));
+        sendCommand(new LeapCommand(Request.getOccupancyGroups()));
+    }
+
+    private void scheduleConnectRetry(long waitMinutes) {
+        logger.debug("Scheduling connection retry in {} minutes", waitMinutes);
+        connectRetryJob = scheduler.schedule(this::connect, waitMinutes, TimeUnit.MINUTES);
+    }
+
+    private synchronized void disconnect() {
+        logger.debug("Disconnecting");
+
+        Thread senderThread = this.senderThread;
+        Thread readerThread = this.readerThread;
+
+        if (connectRetryJob != null) {
+            connectRetryJob.cancel(true);
+        }
+        if (keepAlive != null) {
+            keepAlive.cancel(true);
+        }
+
+        // May be called from keepAliveReconnect thread, so call cancel with false
+        reconnectTaskCancel(false);
+
+        if (senderThread != null && senderThread.isAlive()) {
+            senderThread.interrupt();
+        }
+        if (readerThread != null && readerThread.isAlive()) {
+            readerThread.interrupt();
+        }
+        if (sslsocket != null) {
+            try {
+                sslsocket.close();
```

Comment:
In order to avoid the null checker warnings cache `this.sslsocket` to a local variable and perform your logic on that local variable instead.

Reply:
None of these four null checker warnings were showing up in Eclipse.

Reply:
They might not show up in Eclipse but they do show up in the build console:
```
[WARNING] /home/travis/build/openhab/openhab-addons/bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/LeapBridgeHandler.java:[588,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] /home/travis/build/openhab/openhab-addons/bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/LeapBridgeHandler.java:[595,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] /home/travis/build/openhab/openhab-addons/bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/LeapBridgeHandler.java:[602,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] /home/travis/build/openhab/openhab-addons/bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/LeapBridgeHandler.java:[659,44] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] /home/travis/build/openhab/openhab-addons/bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/LeapBridgeHandler.java:[660,21] Redundant null check: The variable buttonList cannot be null at this location
```

Reply:
Yes, I'm just wondering why they don't show up in Eclipse as well. Should we open a an issue on it? And if so, where?

Reply:
I'm not really sure.

Reply:
These are all fixed.